### DISCUSSION TITLE
Complete n00b, but OMG

### DISCUSSION THREAD

**Discussion by malaki86** [nebhead/PiFire#157] [Q&A]
I've always wanted a GOOD pid controller for my smokers, but at $200+ each, and usually without WiFi... I did find a great deal on a Camp Chef gen 3 (large screen with rotary button input) and installed it on one of my Pit Boss verticals. It's great, but lacks in features. 

I found PiFire yesterday and it seems absolutely fantastic. I love the DIY aspect of it . I've built Arduino-based fan controllers, but the only thing that I have ever used a Rpi for is to be small servers.

I saw that it supports the ADS1000s, and the MAX31865, but how hard would it be to incorporate the MCP9600 K-type thermocouple, possibly even 2 for the averaging? They're I2C devices, meaning that you don't have to worry about the SPI interface with them. This would also allow for all 4 channels of the initial ADS1115 to be used for food probes.

In the documents, I see that PiFire supports rotary encoders, but absolutely nothing about the connections needed for it. Maybe add a new socket on the main board for a standard rotary button to be easily plugged in? Similar to this, the documentation says that you can easily add additional ADS1115 ADCs, but nothing about changing the ADC I2C address. 

Screen - what's a decent screen in the 3" or larger? My eyesight is horrible and the 1-2" screens are worthless. Honestly, I would put a 5-7" on it if I knew how LOL.

Schematics - is there any easy way to load the schematics into KiCad without having to basically start from scratch? I would love to take a shot at designing my own PCB.

**Comment by nebhead** [comment]
> I found PiFire yesterday and it seems absolutely fantastic. I love the DIY aspect of it . I've built Arduino-based fan controllers, but the only thing that I have ever used a Rpi for is to be small servers.

Welcome to the project!  It's definitely for folks that aren't afraid to dive in and tinker, tune their setups.  The perfect marriage of DIY technology and barbecue.  We have a discord server with lots of great discussion about all of these topics.  
 
> I saw that it supports the ADS1000s, and the MAX31865, but how hard would it be to incorporate the MCP9600 K-type thermocouple, possibly even 2 for the averaging? They're I2C devices, meaning that you don't have to worry about the SPI interface with them. This would also allow for all 4 channels of the initial ADS1115 to be used for food probes.

Looks super simple to add this to the project!  I would venture to say that I could add a module to this, but maybe not move it to production until I'm able to get my hands on one.  

> In the documents, I see that PiFire supports rotary encoders, but absolutely nothing about the connections needed for it. Maybe add a new socket on the main board for a standard rotary button to be easily plugged in? Similar to this, the documentation says that you can easily add additional ADS1115 ADCs, but nothing about changing the ADC I2C address.

Both of these are documented better in the configuration wizard that runs on the device.   But you're right, I probably should add the information in the online documentation.  The encoder uses the same pins as the buttons.  Encoder - CLK = GPIO16, DT = GPIO20, SW = GPIO21, GND = GND, VCC/+ = 3.3V

> Screen - what's a decent screen in the 3" or larger? My eyesight is horrible and the 1-2" screens are worthless. Honestly, I would put a 5-7" on it if I knew how LOL.

I use an ILI9341 3" display and agree I don't like having too small of a display.  I'm also working on the 800x480 5-7" touch screen for PiFire, but still in the early stages.  Making good progress, but not ready for prime-time yet.  I'd say look around on AliExpress for a 3" or larger ILI9341 320x240 display.  You might have to make a custom enclosure if you get something that is differently sized.  

> Schematics - is there any easy way to load the schematics into KiCad without having to basically start from scratch? I would love to take a shot at designing my own PCB.

Unfortunately, it's a bit difficult to export to KiCad.  I think it's possible with certain tools online, but I haven't tried.  You're definitely welcome to give it a try.  I myself was thinking of moving to KiCad for the next version, as I've been dabbling there.

**Comment by malaki86** [comment]
I was working on my own project last year, which had "expandability" built in. It was going to built with the "standard" breakouts for both the ADS1115 & MCP9600, with the ability to have up to 2 of each (8 food & 2 pit probes). Basically, I was going to hardwire the main circuit board to connect the jumpers for the 2nd of each type of board. This would completely eliminate the solder-pad work on the addon boards. Solder the header on, plug it in, done. 

The ESP32 would then check the addresses at startup to detect whether or not a board was at a specific address, and if so, configure the probes.

**Reply by nebhead** [reply]
Having these devices I2C is great because you can expand them to many different inputs.  PiFire is really only limited by the I2C addressing and physical GPIOs.

**Comment by malaki86** [comment]
I saw a discussion about adding touchscreen ability, so would it he worthwhile to get a screen with it built in? ILI9341 4" 480x320 with touch is $1 more than without.

**Reply by nebhead** [reply]
So... I should warn you that 480x320 isn't supported yet so that would be a no-go on that display.  Only the 320x200 display is supported at the moment.  

Also, the touch isn't enabled either (since this requires additional SPI), and there isn't a good python library that is standalone for that chip.  At least that I've found.  Anyway, you might be limited to the 3" display for now.

**Reply by malaki86** [reply]
I'm not seeing any 320x200. Did you mean 320x240? This is one example that I found:

https://a.aliexpress.com/_mLuHzT8

**Comment by nebhead** [comment]
Oh yes 320x240.
